### PR TITLE
chore: add label `build-image-on-pr` to trigger a build

### PR DESCRIPTION
**Pull request by tabVersion**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

as title, a new label to trigger a workflow to build a new image on the current branch, and labeled as the branch name. 

note that the image tag is sanitized: `feature/new-thing` becomes `feature-new-thing`

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: ci/workflows/pull-request.yml [hunk 1]
```diff
@@ -333,6 +333,25 @@ steps:
     timeout_in_minutes: 10
     retry: *auto-retry
 
+  - label: "build-image-on-pr"
+    key: "build-image-on-pr"
+    if: build.pull_request.labels includes "build-image-on-pr" || build.env("CI_STEPS") =~ /(^|,)build-image-on-pr?(,|$$)/
```
Comment (ClSlaid) [inline ci/workflows/pull-request.yml:338]:
```suggestion
    if: build.pull_request.labels includes "build-image-on-pr" || build.env("CI_STEPS") =~ /(^|,)build-image-on-pr(,|$$)/
```

**Review by ClSlaid — APPROVED** [review]

File: ci/workflows/pull-request.yml (hunk 1 above)
Comment (ClSlaid) [reply ci/workflows/pull-request.yml:338]: You might have tried to write the regex as "...sink_tests?(,|$$)/"
However, the "?" here means the "s" in "sink_test[s]" is optional.
So please remove this "?"

File: ci/workflows/pull-request.yml [hunk 2]
```diff
@@ -333,6 +333,25 @@ steps:
     timeout_in_minutes: 10
     retry: *auto-retry
 
+  - label: "build-image-on-pr"
+    key: "build-image-on-pr"
+    if: build.pull_request.labels includes "build-image-on-pr" || build.env("CI_STEPS") =~ /(^|,)build-image-on-pr(,|$$)/
```
Comment (kwannoel) [inline ci/workflows/pull-request.yml:338]:
```suggestion
    if: build.pull_request.labels includes "ci/pr/build-docker-image" || build.env("CI_STEPS") =~ /(^|,)build-image-on-pr(,|$$)/
```

So the naming convention is consistent.

**Comment by kwannoel** [comment]
You can also add the   github label to this PR, to test it works.

File: ci/workflows/pull-request.yml [hunk 3]
```diff
@@ -333,6 +333,25 @@ steps:
...
 
+  - label: "build-image-on-pr"
+    key: "build-image-on-pr"
+    if: build.pull_request.labels includes "ci/pr/build-image-on-pr" || build.env("CI_STEPS") =~ /(^|,)ci\/pr\/build-image-on-pr(,|$$)/
+    command: |
+      # Sanitize branch name for Docker tag
+      export IMAGE_TAG=$(echo "$BUILDKITE_BRANCH" | sed 's/[^a-zA-Z0-9._-]/-/g')
+      echo "publishing new image $IMAGE_TAG from branch $BUILDKITE_BRANCH ($BUILDKITE_COMMIT)"
+      # Trigger docker.yml pipeline
+      buildkite-agent pipeline upload ci/workflows/docker.yml
```
Comment (BugenZhao) [inline ci/workflows/pull-request.yml:344]: This is to build a **CI** image. Is this intended behavior?

Comment (tabVersion) [reply ci/workflows/pull-request.yml:344]: image

I think it is the right workflow.

Comment (BugenZhao) [reply ci/workflows/pull-request.yml:344]: You're right.

**Review by BugenZhao — COMMENTED** [review]
Some concerns:

- It seems that we build the image directly based on the commits in the branch, instead of merging it into the main branch first. I'm afraid there can be confusion when different base commits are used.
- What if the workflow is triggered multiple times on the same branch, given that we only encode the branch name into the image tag.

**Pull request closed without merging**